### PR TITLE
Fixes #30144 - use media_type check not content_type

### PR DESCRIPTION
**Pull request by ezr-ondrej**
It should be the same ATM, but media_type should always stay only media_type, content_type can have more information. Maybe even some clients already may send such content_type headers.

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
[test katello]

File: app/views/api/v2/errors/unsupported_media_type.json.rabl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+object false
+
+node(:message) { _("Media type in 'Content-Type: %s' is unsupported in API v2 for POST and PUT requests. Please use 'Content-Type: application/json'.") % request.content_type }
```
Comment (ares) [inline app/views/api/v2/errors/unsupported_media_type.json.rabl:3]: should this be a media type at the end of the line too?

**Review by ares — CHANGES_REQUESTED** [review]
For some reason, I'm unable to trigger the unsupported content-type error, but that may be something on my side. I thought `curl -v https://admin:changeme@foreman.example.com/api/v2/users -H "Content-Type: text/plain"` would do the trick, but it does not...

One question though.

File: app/views/api/v2/errors/unsupported_media_type.json.rabl (hunk 1 above)
Comment (ezr-ondrej) [reply app/views/api/v2/errors/unsupported_media_type.json.rabl:3]: No, it should not, the header should be still printed out whole in the error message I believe (`content_type` should include )
Or we can rephrase this to `Media type '%s' is unsupported`

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
Katello failures taken care of in Katello/katello#8811

File: app/controllers/api/v2/base_controller.rb [hunk 2]
```diff
@@ -144,9 +144,9 @@ def setup_has_many_params
         append_array_of_ids(params)             # unwrapped params
       end
 
-      def check_content_type
```
Comment (tbrisker) [inline app/controllers/api/v2/base_controller.rb:147]: Should this method be deprecated and call the new method to avoid breaking plugins (as we can see is the case for katello)?

Comment (ezr-ondrej) [reply app/controllers/api/v2/base_controller.rb:147]: hmm it's not as easy :thinking: I can do that on method basis, but with the `before_action`, I'll eighter:
- use the old method, but we will trigger our own deprecation callback,
- use new one, but it won't help the katello at all having the old method still around :disappointed:

The renaming is not necessary and I can keep the old name, but I don't like the unability of renaming callback, how should we do it properly?

Comment (tbrisker) [reply app/controllers/api/v2/base_controller.rb:147]: hmm... this is indeed tricky, adding a deprecation will work for `before_action` usage elsewhere, but not for `skip_before_action`.
How about overriding `self.skip_before_action` to rename it if needed and warn of deprecation?

Comment (ezr-ondrej) [reply app/controllers/api/v2/base_controller.rb:147]: Right :+1: I've done that ^_^

**Comment by tbrisker** [comment]
Looks like katello test failure is related `ArgumentError: Before process_action callback [:setup_has_many_params] has not been defined`

**Comment by ezr-ondrej** [comment]
> Looks like katello test failure is related `ArgumentError: Before process_action callback [:setup_has_many_params] has not been defined`

Stupid me, I've forgotten `*` in `super(*names)`

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !
